LPS-174471 + LPS-174465 Site template friendly URL conflict warnings on page trees

vendeltoreki opened the pull request:
Hi @liferay-echo ,

As discussed, I'm resending the pull request with only the back-end changes: https://github.com/liferay-echo/liferay-portal/pull/11997 .

These are code changes that adds the friendly URL conflict information into the JSON objects:
* Site Builder (MillerColumns): https://github.com/liferay-echo/liferay-portal/commit/f78251c256283cbf6be190fe08006f000727018a
* Page Tree: https://github.com/liferay-echo/liferay-portal/commit/29dfe1078259b35f85c91b74b349def58042c7cd

I'm not sure what would be the best way to format this data for your front-end implementation. Please let me know if you had something else in mind.

Feature flag: `feature.flag.LPS-174471=true`

cc: @ealonso , @noraszel

Thanks,
Vendel

liferay-echo commented:
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

victorg1991 commented on modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/util/LayoutsTreeImpl.java:
```diff
@@ -443,6 +478,25 @@ private JSONObject _toJSONObject(
 			}
+		).put(
+			"title",
+			() -> {
```
The same way that we need in the miller columns, can we just pass here a boolean prop `urlConflict` that is true when the layout has some url conflicts?

Also, we need to pass the `siteTemplate` prop here: https://github.com/liferay/liferay-portal/blob/1b0dd8add155c5cb70c65190d8cafa6ab536bc76/modules/apps/staging/staging-taglib/src/main/java/com/liferay/staging/taglib/internal/display/context/LayoutsTreeDisplayContext.java#L187-L189 when the group is a site template to show a different message in that case

victorg1991 commented on modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/util/LayoutsTreeImpl.java:
```diff
@@ -401,6 +429,13 @@ private JSONObject _toJSONObject(
+						layout.getName(themeDisplay.getLocale()),
+						StringPool.SPACE, StringPool.OPEN_PARENTHESIS,
+						StringPool.EXCLAMATION, StringPool.CLOSE_PARENTHESIS);
+				}
```
This is not needed since we are not going to show the exclamation mark anymore

victorg1991 commented:
**I didn't review all the backend code** , just reviewed the part we need to contribute the front to this PR

@vendeltoreki could you check out my comments and let me know if you need further information? :)

vendeltoreki commented:
Hi @victorg1991 ,

Thanks for the detailed review. I rebased the conflicting `LayoutsTreeImpl`, and added the requested changes to `MillerColumnsDisplayContext`.

Do you think we should rather use the string `isSiteTemplate` instead of `siteTemplate` in the JSON to make it look more like a boolean? Or possibly `isLayoutSetPrototype`, since we call site templates "LayoutSetPrototype" everywhere in the code?

Thanks,
Vendel

victorg1991 commented:
Thanks @vendeltoreki  :)

> Hi @victorg1991 ,
> 
> Thanks for the detailed review. I rebased the conflicting LayoutsTreeImpl, and added the requested changes to MillerColumnsDisplayContext.
> 
> Do you think we should rather use the string isSiteTemplate instead of siteTemplate in the JSON to make it look more like a boolean? Or possibly isLayoutSetPrototype, since we call site templates "LayoutSetPrototype" everywhere in the code?
> 
> Thanks,
> Vendel
> 

Yeah I think that maybe an improvement :) but since we are a bit in a hurry with this and the pr seems to be big enough, maybe we can change it in a future pr after the front is done? what do you think?

vendeltoreki commented:
> Thanks @vendeltoreki :)
> 
> > Hi @victorg1991 ,
> > Thanks for the detailed review. I rebased the conflicting LayoutsTreeImpl, and added the requested changes to MillerColumnsDisplayContext.
> > Do you think we should rather use the string isSiteTemplate instead of siteTemplate in the JSON to make it look more like a boolean? Or possibly isLayoutSetPrototype, since we call site templates "LayoutSetPrototype" everywhere in the code?
> > Thanks,
> > Vendel
> 
> Yeah I think that maybe an improvement :) but since we are a bit in a hurry with this and the pr seems to be big enough, maybe we can change it in a future pr after the front is done? what do you think?

I think it's ok, we can change it later

vendeltoreki commented:
Hi @victorg1991 ,

No worries, I managed to rebase, and update the right file :)

Thanks,
Vendel

victorg1991 commented:
Many thanks!! :)

ealonso commented:
Hey @vendeltoreki @ruben-pulido,

I'll take care of it.

vendeltoreki commented:
Hi @ruben-pulido , @ealonso ,

I pushed 3 commits to the branch, adding the requested changes.

Thanks,
Vendel
